pinot-segment-local/src/main/java/org/apache/pinot/segment/local/io/compression/GzipCompressor.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.segment.local.io.compression;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.zip.Deflater;
+import org.apache.pinot.segment.spi.compression.ChunkCompressionType;
+import org.apache.pinot.segment.spi.compression.ChunkCompressor;
+
+
+/**
+ * Implementation of {@link ChunkCompressor} using GZIP compression algorithm.
+ */
+class GzipCompressor implements ChunkCompressor {
+
+  static final ThreadLocal<GzipCompressor> INSTANCE = ThreadLocal.withInitial(GzipCompressor::new);
+
+  private final Deflater _compressor;
+
+  private GzipCompressor() {
+    _compressor = new Deflater();
```

Comment:
Is it expensive to create a `Deflater`? If not, I think we can simply create one each time without using thread local. The problem of using thread local is that the memory used by `Deflater` is not released

Reply:
done in 69e09577728ed6a19d47eef39af1e8bed7cd7a1c